### PR TITLE
Bluetooth: Mesh: Fix overflow on lvl_delta

### PR DESCRIPTION
**Pull request by trond-snekvik**
When performing a delta set in the lighting models, the delta is added
to the current value, then stored in an integer that only holds 16 bits.
If the delta change overflows the 16 bit integer, it will wrap around
when stored, instead of clamping to the min/max of the storage variable.
Although clamping is done in the set-functions, it's already too late,
as the level has overflowed its variable and wrapped around.

Adds boundary clamp before storing the target values in the 16 bit
integer for gen_plvl, lightness_srv and light_temp_srv.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/mesh/gen_plvl_srv.c [hunk 1]
```diff
@@ -447,16 +447,21 @@ static void lvl_delta_set(struct bt_mesh_lvl_srv *lvl_srv,
 	struct bt_mesh_plvl_srv *srv =
 		CONTAINER_OF(lvl_srv, struct bt_mesh_plvl_srv, lvl);
 	struct bt_mesh_plvl_status status = { 0 };
-
-	uint16_t start_value = srv->last;
+	uint16_t start_value;
```
Comment (dilner) [inline subsys/bluetooth/mesh/gen_plvl_srv.c:450]: One thing that confuses me is the use of start and start_value. Is there any difference between them. Also I think "value" should be avoided if it's not generic, if there's no reason I'm unaware of. Maybe change to start_lvl to differentiate from temperature etc.

File: subsys/bluetooth/mesh/light_temp_srv.c [hunk 2]
```diff
@@ -189,23 +189,22 @@ static void lvl_delta_set(struct bt_mesh_lvl_srv *lvl_srv,
 		CONTAINER_OF(lvl_srv, struct bt_mesh_light_temp_srv, lvl);
 	struct bt_mesh_light_temp_status status = { 0 };
 	struct bt_mesh_light_temp_cb_set cb_msg;
-	int16_t start_value = temp_to_lvl(srv, srv->temp_last);
-	int16_t target_value;
+	int16_t start, target;
```
Comment (dilner) [inline subsys/bluetooth/mesh/light_temp_srv.c:192]: Everything I said that applies to start also applies to target. Target_lvl maybe?

**Review by dilner — CHANGES_REQUESTED** [review]
Minor change of variable names suggested.

File: subsys/bluetooth/mesh/gen_plvl_srv.c (hunk 1 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/gen_plvl_srv.c:450]: There's no difference, so it's an unnecessary inconsistency. I'll change it.

File: subsys/bluetooth/mesh/light_temp_srv.c (hunk 2 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/light_temp_srv.c]: Changed. Notice that in lightness_srv, we're using `target_actual`, as this target value is using the "Lightness Actual" representation.

**Review by dilner — APPROVED** [review]

**Review by ludvigsj — APPROVED** [review]
Looks okay to me

**Pull request merged**